Storage/Processor/Processor.php

```diff
@@ -207,26 +207,39 @@ private function generateImage(Config $config, string $filename = null): string
         return $generatedImage;
     }
 
+    private function getResourceFromAsset(Config $config)
+    {
+        if ($config->getFilename()) {
+            return fopen($config->getFilename(), 'r');
+        }
 
-    public function getResource(Config $config, string $filename)
+        return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
+    }
+
+    public function getResource(Config $config, string $filename, bool $noCache = false)
     {
-        $cacheableResult = $config->cacheableResult();
         if (!$config->getStorageContext()) {
-            if ($config->getFilename()) {
-                return fopen($config->getFilename(), 'r');
-            }
+            return $this->getResourceFromAsset($config);
+        }
 
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getStorageContext());
+        if ($config->isCacheableResult() && !$noCache) {
+            try {
+                return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
+            } catch (NotFoundException $e) {
+            } catch (\Exception $e) {
+                $this->logger->warning('log.unexpected_exception', ['error_message' => $e->getMessage()]);
+            }
         }
 
-        try {
-            return $this->storageManager->getResource($config->getAssetHash(), $config->getConfigHash());
-        } catch (NotFoundException $e) {
-            $generatedResource = $this->generateResource($config);
-            if ($cacheableResult) {
+        $generatedResource = $this->generateResource($config);
+
+        if ($config->isCacheableResult()) {
+            try {
```

Comment:
Why not one big try catch that contains all the ifs?

Reply:
The fact that we are not able to retrieve a cached version or not able to save in cache is not a big deal as soon as we are still able to generate a response. Just a log in that case. If we aren’t able to generate the response it’s a 500 error.

Reply:
Yeah but higher in this function there is another try catch. Can’t they be combined?

Reply:
it has been refactored with getResourceFromCache and saveResourceToCache functions without nested try/if